### PR TITLE
Improve: refactor caliper app

### PR DESCRIPTION
**Pull request by asamolion**
Firstly, the base_transformer for caliper has been refactored into
multiple procedures rather than a class. Furthermore, The
CaliperProcessor class has been made compatible with both tracking
backends of Open edX and is completely configurable

**Checks before merge:**

- [x] Reviewed
- [x] Commits squashed

### CODE REVIEW

File: openedx/features/caliper_tracking/base_transformer.py [hunk 1]
```diff
@@ -8,73 +8,80 @@
...
 
-class CaliperBaseTransformer(object):
-    """Base transformer class
-    This class is responsible for adding all the caliper compliant
-    fields which are common to all events
+"""Base transformer class
+This class is responsible for adding all the caliper compliant
+fields which are common to all events
+"""
+
```
Comment (imhassantariq) [inline openedx/features/caliper_tracking/base_transformer.py]: Please remove this doc string

Comment (asamolion) [reply openedx/features/caliper_tracking/base_transformer.py]: Done.

**Comment by danialmalik** [comment]
Events are not being transformed. Please look into that. 

And how is value of `TRACKING_BACKENDS` (used by `Track` app) being handled now ? I cannot see that in file changes.

**Comment by asamolion** [comment]
@danialmalik This PR does not contain the configuration. Each developer will have to do that separately on his/her own machine.

**Comment by asamolion** [comment]
Also, the value of `TRACKING_BACKENDS` is handled by edx code. Not ours.

**Comment by danialmalik** [comment]
In that case, not all of the events are transformed. Unless we set the value for `TRACKING_BACKENDS`.

File: openedx/features/caliper_tracking/processor.py [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
             related_function = EVENT_MAPPING[event.get('event_type')]
             return related_function(event, caliper_event)
         except KeyError as ex:
             logger.exception("Missing transformer method implementation for %s" % event.get('event_type'))
         except Exception as ex:
             logger.exception(ex.args)
-            pass
+
+    def send(self, event):
+        logger.info(self.__call__(event))
```
Comment (danialmalik) [inline openedx/features/caliper_tracking/processor.py]: Currently page visit events are also being transformed therefore this check is needed i think.

```suggestion
        if not event['event_type'].startswith('/'):
            logger.info(self.__call__(event))
```

Comment (asamolion) [reply openedx/features/caliper_tracking/processor.py]: Why do we need this?

Comment (asamolion) [reply openedx/features/caliper_tracking/processor.py]: Added.

**Comment by mumarkhan999** [comment]
Manual QA done. :+1:

**Pull request merged**